google/cloud/documentai_toolbox/converters/config/converter_helpers.py

```diff
@@ -0,0 +1,485 @@
+            annotated_bytes=blobs[0],
+            document_bytes=blobs[1],
+            schema_bytes=blobs[2],
+            project_id=project_id,
+            location=location,
+            processor_id=processor_id,
+            retry_number=1,
+            name=blobs[3],
+        )
+
+        if docproto == None:
+            did_not_convert.append(f"{output_prefix}/{blobs[3]}")
+            continue
+
+        for entity in docproto.entities:
+            if entity.type_ not in unique_types:
+                unique_types.append(entity.type_)
+
+        files[blobs[3]] = str(documentai.Document.to_json(docproto))
+
+    return files, unique_types, did_not_convert
+
+
+def _upload(files, gcs_output_path):
+    match = re.match(r"gs://(.*?)/(.*)", gcs_output_path)
+
+    if match is None:
+        raise ValueError("gcs_prefix does not match accepted format")
+
+    output_bucket, output_prefix = match.groups()
+
+    if output_prefix == None:
+        output_prefix = "/"
+
+    file_check = re.match(r"(.*[.].*$)", output_prefix)
+
+    if file_check is not None:
+        raise ValueError("gcs_prefix cannot contain file types")
+
+    download_pool = futures.ThreadPoolExecutor(10)
+    uploads = []
+    print("-------- Uploading Started --------")
+    for i, key in enumerate(files):
+        op = output_prefix.split("/")
+        op.pop()
+        prefix = "/".join(op)
+        if "config" not in key and "annotations" not in key:
+            _upload = download_pool.submit(
+                _upload_file,
+                output_bucket,
+                f"{output_prefix}/{key}.json",
+                files[key],
+            )
+            uploads.append(_upload)
+
+    futures.wait(uploads)
+
+
+def convert_documents_with_config(
+    gcs_input_path: str,
+    gcs_output_path: str,
+    project_id: str,
+    location: str,
+    processor_id: str,
+    config_path: str = None,
+) -> None:
+    r"""Converts all documents in gcs_path to docproto.
+
+    Args:
+        gcs_input_path (str):
+            Required. The gcs path to the folder containing all non docproto documents.
+
+            Format: `gs://{bucket}/{optional_folder}`
+        gcs_output_path (str):
+            Required. The gcs path to the folder to upload the converted docproto documents to.
+
+            Format: `gs://{bucket}/{optional_folder}`
+        project_id (str):
+            Required.
+        location (str):
+            Required.
+        processor_id (str):
+            Required.
+
+    Returns:
+        None.
+
+    """
+    match = re.match(r"gs://(.*?)/(.*)", gcs_input_path)
+
+    if match is None:
+        raise ValueError("gcs_prefix does not match accepted format")
+
+    output_bucket, output_prefix = match.groups()
+
+    if output_prefix == None:
+        output_prefix = "/"
+
+    file_check = re.match(r"(.*[.].*$)", output_prefix)
+
+    if file_check is not None:
+        raise ValueError("gcs_prefix cannot contain file types")
```

Comment:
Looks like this reuses the code from `upload` could be good to make another function